client/packages/inventory/src/Stocktake/context/stocktakeLineError.tsx

```diff
@@ -0,0 +1,63 @@
+import { PropsWithChildrenOnly, RecordWithId } from '@common/types';
+import React, { createContext, useContext, useState } from 'react';
+import {
+  AdjustmentReasonNotProvidedErrorFragment,
+  AdjustmentReasonNotValidErrorFragment,
+  SnapshotCountCurrentCountMismatchErrorFragment,
+  StockLineReducedBelowZeroErrorFragment,
+} from '../api';
+
+export type StocktakeLineError =
+  | AdjustmentReasonNotProvidedErrorFragment
+  | StockLineReducedBelowZeroErrorFragment
+  | AdjustmentReasonNotValidErrorFragment
+  | SnapshotCountCurrentCountMismatchErrorFragment;
+
+const useStocktakeLineErrors = () => {
+  const [errors, setErrors] = useState<{
+    [id: string]: StocktakeLineError | undefined;
+  }>({});
+
+  const getError = ({ id }: RecordWithId): StocktakeLineError | undefined => {
+    return errors[id];
+  };
+
+  const setError = (id: string, error: StocktakeLineError) => {
+    setErrors(errors => ({ ...errors, [id]: error }));
+  };
+
+  const unsetError = (id: string) => {
+    setErrors(errors => ({ ...errors, [id]: undefined }));
+  };
+
+  const unsetAll = () => {
+    setErrors({});
+  };
+
+  return { errors, setError, setErrors, getError, unsetError, unsetAll };
+};
+export type UseStocktakeLineErrors = ReturnType<typeof useStocktakeLineErrors>;
+
+const StocktakeLineErrorContext = createContext<UseStocktakeLineErrors | null>(
+  null
+);
+
+export const useStocktakeLineErrorContext = () => {
+  const context = useContext(StocktakeLineErrorContext);
+
+  if (!context) throw new Error('Context does not exist');
```

Comment:
This is safe right ? rather then defining defaults for each of the methods

Reply:
I'm not sure why the context wouldn't be valid 🤷  
throwing an error here would just give the fallback 'Oops! something's gone wrong' screen.. so is ok